src/TreeView/TreeView.tsx

```diff
@@ -241,9 +241,57 @@ const UlBox = styled.ul<SxProp>`
 const Root: React.FC<TreeViewProps> = ({'aria-label': ariaLabel, 'aria-labelledby': ariaLabelledby, children}) => {
   const containerRef = React.useRef<HTMLUListElement>(null)
   const [ariaLiveMessage, setAriaLiveMessage] = React.useState('')
+  const announceUpdate = React.useCallback((message: string) => {
+    setAriaLiveMessage(message)
+  }, [])
 
-  useRovingTabIndex({containerRef})
+  /**
+   * Manage the active element when the user presses `PageUp` or `PageDown`
+   * while using the TreeView
+   */
+  function onKeyDown(event: React.KeyboardEvent) {
```

Comment:
Can we move this code into the useRovingTabIndex() hook? Something like this:

```diff
diff --git a/src/TreeView/useRovingTabIndex.ts b/src/TreeView/useRovingTabIndex.ts
index 83d336645..9903442ac 100644
--- a/src/TreeView/useRovingTabIndex.ts
+++ b/src/TreeView/useRovingTabIndex.ts
@@ -5,7 +5,12 @@ export function useRovingTabIndex({containerRef}: {containerRef: React.RefObject
   // TODO: Initialize focus to the aria-current item if it exists
   useFocusZone({
     containerRef,
-    bindKeys: FocusKeys.ArrowVertical | FocusKeys.ArrowHorizontal | FocusKeys.HomeAndEnd | FocusKeys.Backspace,
+    bindKeys:
+      FocusKeys.ArrowVertical |
+      FocusKeys.ArrowHorizontal |
+      FocusKeys.HomeAndEnd |
+      FocusKeys.Backspace |
+      FocusKeys.PageUpDown,
     preventScroll: true,
     getNextFocusable: (direction, from, event) => {
       if (!(from instanceof HTMLElement)) return
@@ -67,6 +72,14 @@ export function getNextFocusableElement(activeElement: HTMLElement, event: Keybo
     case 'End':
       // Focus last visible element
       return getLastElement(activeElement)
+
+    case 'PageUp':
+      // TODO
+      return
+
+    case 'PageDown':
+      // TODO
+      return
   }
 }
```

Reply:
@colebemis will adding it to `bindKeys` cause `event.preventDefault()` to be called? One thing that's nice about the approach here is that `event.preventDefault()` does not need to be called so we can tag-a-long with the native `PageDown` or `PageUp` functionality.

Unfortunately if the page scroll is not being updated then the heuristic will need to be changed a bit in order to work. Let me know what you think

Reply:
Ah bummer, you're right. How do you feel about pulling this code out into a `usePageUpDown` hook?